node/src/chain/acceptor.rs

```diff
@@ -228,6 +229,33 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
                 Ok(txs)
             })?;
 
+            for iter in 0..blk.header().iteration {
+                let mut prov = provisioners_list.clone();
```

Comment:
Duplicating the provisioners_list for every iteration could be inefficient, especially if a block originates from a higher iteration